solr/core/src/java/org/apache/solr/filestore/DistribFileStore.java

```diff
@@ -428,10 +428,10 @@ public boolean fetch(String path, String from) {
   @Override
   public void get(String path, Consumer<FileEntry> consumer, boolean fetchmissing)
       throws IOException {
-    File file = getRealpath(path).toFile();
-    String simpleName = file.getName();
+    Path file = getRealpath(path);
+    String simpleName = file.getFileName().toString();
     if (isMetaDataFile(simpleName)) {
-      try (InputStream is = new FileInputStream(file)) {
+      try (InputStream is = new FileInputStream(file.toString())) {
```

Comment:
Could we ditch the `FileInputStream`?   I asked GPT, and it said:

```
InputStream inputStream = Files.newInputStream(path))
```

The recommended approach is using `Files.newInputStream()` because:
- It's more flexible (supports options)
- Works with any filesystem provider
- Better handles symbolic links
- Part of the modern NIO.2 API

Reply:
cool